src/videojs-hls.js

```diff
@@ -679,7 +718,7 @@ videojs.Hls.prototype.fillBuffer = function(offset) {
   // being buffering so we don't preload data that will never be
   // played
   if (!this.playlists.media().endList &&
-      !this.player().hasClass('vjs-has-started') &&
+      !player.hasClass('vjs-has-started') &&
```

Comment:
I believe we'll be switching all of the `vjs-has-started` to a mechanism like `hasPlayed_` with the video.js 5.x update. I was trying to minimize the surface area of this particular PR.